Add loading finished event

assertchris opened the pull request:
This PR includes some fixes for multiple `unmount()` calls, and an event which triggers when the file is finished loading. It should address this issue: https://github.com/johnsonsu/react-native-sound-player/issues/28

The commit history isn't great, but I'm submitting this early to see whether or not you approve of the implementation, before I go through the hassle of cleaning up the commit history and reverting style-only changes my editor inadvertently made to `index.js`...

While I have your attention: thank you for making this. I have found it useful, and am about to submit an app to both stores which includes this library.

johnsonsu commented:
@assertchris Thank you very much for this PR! This looks great!

I think for this event listener to be useful, we also need a new function, maybe `mountFile()` that only does the mounting of the the file but not the playing. Then when the user call `play()`, it will play the file immediately without any delay. But this PR is 90% of the feature! Hopefully I will get sometime this weekend to complete this feature and do a release!

Thank you very much for your support and I hope a great success for your app!

assertchris commented:
Great! I've reverted prettier formatting to `index.js`, and cleaned up the commit history. It's a force push, so you'll need to check the branch out again if you already checked it out. I can try to add the `mount()` function myself, if you like; but I'm not 100% sure I'll get it right... 😄
